genesis-programs/src/lib.rs

```diff
@@ -49,102 +49,96 @@ enum Program {
     BuiltinLoader((String, Pubkey, ProcessInstructionWithContext)),
 }
 
-fn get_programs(operating_mode: OperatingMode, epoch: Epoch) -> Option<Vec<Program>> {
+// given operating_mode and epoch, return the entire set of enabled programs
+fn get_programs(operating_mode: OperatingMode, epoch: Epoch) -> Vec<Program> {
+    let mut programs = vec![];
+
     match operating_mode {
         OperatingMode::Development => {
-            if epoch == 0 {
-                // Programs used for testing
-                Some(vec![
-                    Program::BuiltinLoader(solana_bpf_loader_program!()),
-                    Program::BuiltinLoader(solana_bpf_loader_deprecated_program!()),
-                    Program::Native(solana_vest_program!()),
-                    Program::Native(solana_budget_program!()),
-                    Program::Native(solana_exchange_program!()),
-                ])
-            } else if epoch == std::u64::MAX {
+            // Programs used for testing
+            programs.extend(vec![
+                Program::BuiltinLoader(solana_bpf_loader_program!()),
+                Program::BuiltinLoader(solana_bpf_loader_deprecated_program!()),
+                Program::Native(solana_vest_program!()),
+                Program::Native(solana_budget_program!()),
+                Program::Native(solana_exchange_program!()),
+            ]);
+
+            #[allow(clippy::absurd_extreme_comparisons)]
+            if epoch >= std::u64::MAX {
                 // The epoch of std::u64::MAX is a placeholder and is expected
                 // to be reduced in a future network update.
-                Some(vec![Program::BuiltinLoader(solana_bpf_loader_program!())])
-            } else {
-                None
+                programs.extend(vec![Program::BuiltinLoader(solana_bpf_loader_program!())]);
             }
         }
-        OperatingMode::Stable => {
-            if epoch == std::u64::MAX {
+        OperatingMode::Preview => {
+            #[allow(clippy::absurd_extreme_comparisons)]
+            if epoch >= std::u64::MAX {
                 // The epoch of std::u64::MAX is a placeholder and is expected
                 // to be reduced in a future network update.
-                Some(vec![
+                programs.extend(vec![
                     Program::BuiltinLoader(solana_bpf_loader_program!()),
                     Program::Native(solana_vest_program!()),
                 ])
-            } else {
-                None
             }
         }
-        OperatingMode::Preview => {
-            if epoch == std::u64::MAX {
+        OperatingMode::Stable => {
+            // at which epoch, bpf_loader_program is enabled??
```

Comment:
@mvines Do you know at what epoch the bpf loader was added?

Also, @ryoqun At whatever epoch that is ^^ we should be adding `solana_bpf_loader_deprecated_program` because on tip that is the original loader.

Reply:
Looks like the original loader was enabled in epoch 34

https://github.com/solana-labs/solana/blob/86419dfd3c62ca0054c94f5b70578950f2dd922e/genesis-programs/src/lib.rs#L74

@mvines Does this jive with what you would expect?

Reply:
@ryoqun I pushed changes to this to `solana/fragile-programs-gating`